src/main.py

```diff
@@ -209,6 +210,12 @@ def _run_testing(env: BaseEnv, approach: BaseApproach) -> Metrics:
         except (ApproachTimeout, ApproachFailure) as e:
             print(f"Task {i+1} / {len(test_tasks)}: Approach failed to "
                   f"solve with error: {e}")
+            if CFG.make_failure_videos and e.info.get("partial_refinements"):
```

Comment:
Is the idea of this `get` here that "partial_refinements" might not be in the info dict sometimes? When would this happen?

Reply:
@tomsilver need your help understanding this one

Reply:
it could happen if we time out during grounding, or just never get to the point where we start refining, I think?